File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    const visit = [];
    const result = [];

    for(let i = 0; i < s.length; i++){
        if(!visit.includes(s[i])){
            visit.push(s[i]);
            result.push(-1);
        }else{
            result.push(i - visit.lastIndexOf(s[i]));
            visit.push(s[i]);
        }
    }
    return result
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    visit = '';
    result = [];
    
    for i in range(len(s)):
        if s[i] in visit:
            result.append(i - visit.rfind(s[i]));
            visit += s[i];
        else:
            result.append(-1);
            visit += s[i];
    return result
            